### PR TITLE
Rename libvpi to libcocotbvpi for consistency

### LINKED ISSUE #1389: Icarus: Rename VPI library from gpivpi to libvpi for consistency
All other simulators using VPI use a library named "libvpi", for Icarus it's named gpivpi. There seems to be no technical reason for this discrepancy. Rename it for consistency.

### PR DESCRIPTION
**Pull request by themperek**
Resolves #1389

It would be good if someone else checked with other simulators too. @cmarqu

### CODE REVIEW

File: cocotb/share/makefiles/simulators/Makefile.xcelium [hunk 1]
```diff
@@ -79,8 +79,8 @@ endif
 # which does not play nice with our multi language, so we supply GPI_EXTRA=cocotbvhpi if needed.
 
 #GPI_EXTRA:=
+GPI_ARGS = -loadvpi $(LIB_DIR)/libcocotbvpi:vlog_startup_routines_bootstrap
 ifeq ($(TOPLEVEL_LANG),verilog)
-    GPI_ARGS = -loadvpi $(LIB_DIR)/libvpi
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.xcelium]: This looks unrelated?

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: Actually this is very interesting. This line was doing nothing before. For some reason, `libvpi` would be loaded in any case.  Now since we do not have  `libvpi`  have to use this properly.

**Comment by themperek** [comment]
I do not get anything if I run `git ls-files | xargs egrep -l '\s+$'` locally.

File: cocotb/share/makefiles/simulators/Makefile.xcelium (hunk 1 above)
Comment (eric-wieser) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: This type of "very interesting" is what would really be great in the commit message

**Comment by themperek** [comment]
@cmarqu Can you test this on your side?

File: cocotb/share/makefiles/simulators/Makefile.ius [hunk 2]
```diff
@@ -71,8 +71,9 @@ endif
 # which does not play nice with our multi language, so we supply GPI_EXTRA=cocotbvhpi if needed.
 
 GPI_EXTRA:=
+GPI_ARGS = -loadvpi $(LIB_DIR)/libcocotbvpi:vlog_startup_routines_bootstrap
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.ius]: Do you know why we suddenly need the bootstrapping?

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.ius]: No

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.ius]: Before it was somehow automatic because of name libvpi.

Comment (cmarqu) [reply cocotb/share/makefiles/simulators/Makefile.ius]: For me,  it's working without the explicit name of the startup routine in both Xcelium 19 (modified and used both ``Makefile.ius`` and ``Makefile.xcelium``) and Incisive 15 (used ``Makefile.ius``).

File: cocotb/share/makefiles/simulators/Makefile.xcelium [hunk 3]
```diff
@@ -79,8 +79,8 @@ endif
 # which does not play nice with our multi language, so we supply GPI_EXTRA=cocotbvhpi if needed.
 
 #GPI_EXTRA:=
+GPI_ARGS = -loadvpi $(LIB_DIR)/libcocotbvpi:vlog_startup_routines_bootstrap
 ifeq ($(TOPLEVEL_LANG),verilog)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.xcelium]: Any idea why we need the vpi if the top-level language is VHDL?

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: `vhpi` was and is loading via `GPI_EXTRA`. Would need to try other way.

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: If one does ` -loadvpi $(LIB_DIR)/ius/libcocotbvhpi:vhpi_startup_routines_bootstrap`

```
     -.--ns INFO     cocotb.gpi                         ../embed/gpi_embed.c:71   in set_program_name_in_venv        Did not detect Python virtual environment. Using system-wide Python interpreter
     -.--ns INFO     cocotb.gpi                         ../gpi/GpiCommon.cpp:104  in gpi_print_registered_impl       VHPI registered
     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:431  in arm_callback                    VHPI Error level 3: VHPI - Internal Error : Snapshot not found in Registering Callbacks !!
FILE (NULL):0
     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:433  in arm_callback                    VHPI: Unable to register callback a handle for VHPI type vhpiCbStartOfSimulation(1034)
     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:431  in arm_callback                    VHPI Error level 3: VHPI - Internal Error : Snapshot not found in Registering Callbacks !!
FILE (NULL):0
     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:433  in arm_callback                    VHPI: Unable to register callback a handle for VHPI type vhpiCbEndOfSimulation(1035)

....

     -.--ns INFO     cocotb.gpi                         ../embed/gpi_embed.c:71   in set_program_name_in_venv        Did not detect Python virtual environment. Using system-wide Python interpreter
     -.--ns INFO     cocotb.gpi                         ../gpi/GpiCommon.cpp:104  in gpi_print_registered_impl       VHPI registered
     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:439  in arm_callback                    VHPI ERROR: Registered callback isn't enabled! Got 1000

     -.--ns ERROR    cocotb.gpi                         ..vhpi/VhpiCbHdl.cpp:439  in arm_callback                    VHPI ERROR: Registered callback isn't enabled! Got 1000


```

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: There are errors if one does different way.

Comment (cmarqu) [reply cocotb/share/makefiles/simulators/Makefile.xcelium]: Needing VPI for VHDL is some specialty with Cadence apparently, there are some comments about this in the code IIRC.

File: cocotb/share/makefiles/simulators/Makefile.ius (hunk 2 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.ius]: Xcelium/IUS will used default vlog_startup_routines symbol only if vpi library name is `libvpi.so`

File: cocotb/share/makefiles/simulators/Makefile.xcelium [hunk 4]
```diff
@@ -75,12 +75,11 @@ ifneq (,$(findstring timescale,$(EXTRA_ARGS)))
     $(error "Please use COCOTB_HDL_TIMEUNIT and COCOTB_HDL_TIMEPRECISION to specify timescale.")
 endif
 
-# Xcelium loads VPI regardless of what is asked for, then VHPI if asked to,
-# which does not play nice with our multi language, so we supply GPI_EXTRA=cocotbvhpi if needed.
+# Loading VHPI cases error so we load VPI always and supply GPI_EXTRA=cocotbvhpi if needed.
```
Comment (imphil) [inline cocotb/share/makefiles/simulators/Makefile.xcelium]:
```suggestion
# Loading the VHPI library causes an error, so we always load the VPI library and supply
# GPI_EXTRA=cocotbvhpi if needed.
```

File: cocotb/share/makefiles/simulators/Makefile.ius [hunk 5]
```diff
@@ -67,12 +67,12 @@ ifneq (,$(findstring timescale,$(EXTRA_ARGS)))
     $(error "Please use COCOTB_HDL_TIMEUNIT and COCOTB_HDL_TIMEPRECISION to specify timescale.")
 endif
 
-# IUS loads VPI regardless of what is asked for, then VHPI if asked to,
-# which does not play nice with our multi language, so we supply GPI_EXTRA=cocotbvhpi if needed.
+# Loading VHPI cases error so we load VPI always and supply GPI_EXTRA=cocotbvhpi if needed.
```
Comment (imphil) [inline cocotb/share/makefiles/simulators/Makefile.ius]:
```suggestion
# Loading the VHPI library causes an error, so we always load the VPI library and supply
# GPI_EXTRA=cocotbvhpi if needed.
```
